### PR TITLE
Added committee scraper for IN

### PR DESCRIPTION
**Pull request by EhtishamSabir**
Tried openstates it for the first time...

### CODE REVIEW

File: scrapers_next/in/committees.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from spatula import URL, HtmlListPage, SkipItem
+
+from openstates.models import ScrapeCommittee
+
+
+class CommitteeDetail(HtmlListPage):
+    source = URL(
+        "https://tlhgp53g3c.execute-api.us-east-2.amazonaws.com/beta/api/getMembers?committee_id=840b8a47-2a40-4e28-9e2a-68e6c66c9e45&session_lpid=session_2021"
```
Comment (johnseekins) [inline scrapers_next/in/committees.py:8]: We'll need a way to dynamically update the session. Also, clarifying how you found this AWS endpoint that has IN API calls would be helpful.

Comment (EhtishamSabir) [reply scrapers_next/in/committees.py:8]: from this site: 
https://beta.iga.in.gov/2022/committees
currently it works without session.

Comment (EhtishamSabir) [reply scrapers_next/in/committees.py:7]: Added the header currently website not sending any session to API... when they do then we will update.
This API is from their new site as their old website https://iga.in.gov/ will not be available after the middle of CY 2023.

**Comment by johnseekins** [comment]
Please take a look at the linting changes: https://github.com/openstates/openstates-scrapers/actions/runs/4006521485/jobs/6880419142

You can manually run `black` with `poetry run black scrapers_next/` in your local repo.

**Comment by EhtishamSabir** [comment]
> Please take a look at the linting changes: https://github.com/openstates/openstates-scrapers/actions/runs/4006521485/jobs/6880419142
> 
> You can manually run `black` with `poetry run black scrapers_next/` in your local repo.

Thanks run black fixed it...

File: scrapers_next/in/committees.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            if chamber == "senate":
+                chamber = "upper"
+            elif chamber == "house":
+                chamber = "lower"
+            else:
+                chamber = "legislature"
+
+            if "subcommittee" in name.lower():
+                classification = "subcommittee"
+                parent = name
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:58]: Currently, there is only one subcommittee in the Indiana legislature: the "School Funding Subcommittee"
- it is a subcommittee of the parent committee "Appropriations", which you can see [here](https://iga.in.gov/legislative/2023/committees/#:~:text=Appropriations,School%20Funding%20Subcommittee)
- within the JSON data, you'll see that it is the only committee that doesn't have a null value for `'parent_committee_id'`
   - it's `'parent_committee_id'` is `'6d40580d-5f50-4815-8310-12123fb18db4'` which is, in fact, the `'id'` for the committee with `'name': 'Appropriations'`
- this `if` clause needs to be rewritten to use the subcommittee's `'parent_committee_id'` to retrieve the name of the parent committee.
   - the code should work for this and all future cases in which the JSON follows this format

File: scrapers_next/in/committees.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+from spatula import URL, HtmlListPage, SkipItem
+from openstates.models import ScrapeCommittee
+
+
+class CommitteeDetail(HtmlListPage):
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:5]: Change to `HtmlPage` object.

File: scrapers_next/in/committees.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
+from spatula import URL, HtmlListPage, SkipItem
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:1]: Don't import `HtmlListPage`, import `HtmlPage` instead.

File: scrapers_next/in/committees.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                role = each_member["position_lpid"]
+                name = each_member["first_name"] + " " + each_member["last_name"]
+                com.add_member(name, role)
+
+            com.add_source(self.source.url)
+            com.extras["members"] = self.response.json()["members"]
+            return com
+
+
+class CommitteeList(HtmlListPage):
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:23]: Change to `HtmlPage` object.

**Review by chrisyamas — CHANGES_REQUESTED** [review]
Overall, great job, especially for your first spatula scraper.  And well done finding the API.

Once you have made the requested changes, we should be ready to merge this (barring any other issues with the data).

File: scrapers_next/in/committees.py (hunk 4 above)
Comment (EhtishamSabir) [reply scrapers_next/in/committees.py:1]: If I import HtmlPage only then I get this error
ERROR:spatula:found no list pages in module scrapers_next.in.committees

Comment (chrisyamas) [reply scrapers_next/in/committees.py:1]: Then that's fine, import both, and just convert only the `CommitteeDetail` class from `HtmlListPage` to `HtmlPage` object.

This ended up being a special case where you used spatula's Html page scrapers to process a JSON response, when we typically use the `JsonListPage` and `JsonPage` spatula objects for that. It worked fine, so we can move forward with it this way, but in the future:
- Use either the `HtmlListPage` or `JsonListPage` class (depending on whether you're processing HTML or JSON) and use the `process_item()` method to iterate through list of items (and use a selector to define the list of items). Then, pass to a `HtmlPage` or `JsonPage` class for scraping the details from any additional page that needs to be called.

File: scrapers_next/in/committees.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+    def process_page(self):
+        com = self.input
+        if com:
+            for each_member in self.response.json()["members"]:
+                role = str(each_member["position"])
+                if role.lower().startswith("type"):
+                    role = role.replace("type_", "").replace("_", " ").title()
+                    if role.endswith("member") or role.endswith("chair"):
+                        role = role.replace("member", " Member").replace(
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:19]: If role includes "member", then role should be stored as just: "Member".

Currently, it's either getting stored as "Majority Member" or "Minority Member". We will already know whether the individuals are in the majority or the minority due to our other data on party affiliation. So just remove "majority" or "minority" when present in the string.

File: scrapers_next/in/committees.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
+from spatula import URL, HtmlListPage, SkipItem, HtmlPage
+from openstates.models import ScrapeCommittee
+
+
+class CommitteeDetail(HtmlPage):
+    example_source = URL(
+        "https://tlhgp53g3c.execute-api.us-east-2.amazonaws.com/beta/api/getMembers?committee_id=840b8a47-2a40-4e28-9e2a-68e6c66c9e45&session_lpid=session_2023"
+    )
+    name = None
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:9]: You can remove this, it's unnecessary.

**Review by chrisyamas — CHANGES_REQUESTED** [review]
Two small changes, and then we should be good.

**Comment by EhtishamSabir** [comment]
> Two small changes, and then we should be good.
Changes complete.

File: scrapers_next/in/committees.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+                com = ScrapeCommittee(
+                    name=name,
+                    chamber=chamber,
+                    parent=parent,
+                    classification=classification,
+                )
+
+                com.add_source(
+                    self.source.url,
+                    note="API from https://beta.iga.in.gov/2022/committees/",
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:77]: Sorry, I should've caught this earlier, but can you change this note to read `"Committee List API from current beta version of Indiana gov site"`?

File: scrapers_next/in/committees.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    def process_page(self):
+        com = self.input
+        if com:
+            for each_member in self.response.json()["members"]:
+                role = str(each_member["position"])
+                role = role.lower().replace("majority ", "").replace("minority ", "").title()
+                name = each_member["first_name"] + " " + each_member["last_name"]
+                com.add_member(name, role)
+
+            com.add_source(self.source.url)
```
Comment (chrisyamas) [inline scrapers_next/in/committees.py:15]: Can you add a note in this one that reads `"Committee Details API"`?

**Review by chrisyamas — CHANGES_REQUESTED** [review]
Everything else looks great. As soon as you push a commit fixing those two `add_source()` notes, then we should be good to merge.

File: scrapers_next/in/committees.py (hunk 8 above)
Comment (EhtishamSabir) [reply scrapers_next/in/committees.py:77]: yes sure.

**Comment by EhtishamSabir** [comment]
> Everything else looks great. As soon as you push a commit fixing those two `add_source()` notes, then we should be good to merge.

updated notes...

**Review by chrisyamas — APPROVED** [review]
Great work, and thanks for being so quick with the requested changes.

**Comment by johnseekins** [comment]
It seems there are still linting problems...

File: scrapers_next/in/committees.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        if com:
+            for each_member in self.response.json()["members"]:
+                role = str(each_member["position"])
+                role = (
+                    role.lower()
+                    .replace("majority ", "")
+                    .replace("minority ", "")
+                    .title()
+                )
+                name = each_member["first_name"] + " " + each_member["last_name"]
```
Comment (johnseekins) [inline scrapers_next/in/committees.py:17]: Let's be consistent with how we use strings. _Most_ strings in this scraper are f-strings...

Comment (EhtishamSabir) [reply scrapers_next/in/committees.py]: done...

**Review by johnseekins — APPROVED** [review]

**Pull request merged**